### PR TITLE
Add products search

### PR DESCRIPTION
**Pull request by DJA89**
# Add products search

https://trello.com/c/8HIZsFdz/4-as-a-user-i-should-be-able-to-search-for-products

## Description:

Adds search functionality to the site, using the top bar to search for something it'll redirect it to the products page while loading the products using this search

## Notes:

* Saving the search in the url params so it's easy to save it and share it.

### Desktop

image

### Mobile

image

### Desktop without search term

image

### Desktop search with results (notice the url too)

image

### Desktop search without results

image

### Mobile without search term (*chair* is in the search bar from a previous search)

image

### Mobile search with results

image

### Mobile search without results

image

### CODE REVIEW

File: src/components/MobileFilters/MobileFilters.tsx [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+
 export default function MobileFilters() {
-  return <div className="h-19 w-full"></div>;
+  const searchParams = useSearchParams();
+  const search = searchParams?.get('search');
+
+  return (
+    <div className="flex h-19 w-full flex-col px-4 py-3">
+      <div className="text-lg">
+        {search ? <>You searched for {`"${search}"`}</> : <>{'All Products'}</>}
```
Comment (FlaviaCohen) [inline src/components/MobileFilters/MobileFilters.tsx:10]: Do you need these fragments?

**Review by FlaviaCohen — APPROVED** [review]

File: src/app/(authenticated)/products/DesktopProducts.tsx [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
       <div className="flex">
         <DesktopFilters />
-        <div className="flex flex-col">
-          <div className="h-8 w-71"></div>
+        <div className="flex w-full flex-col md:w-210">
+          <div className="w-71 text-2xl">
+            {search ? (
+              <>
+                You searched for{' '}
+                <span className="font-bold">{`"${search}"`}</span>
```
Comment (agustinmrgo) [inline src/app/(authenticated)/products/DesktopProducts.tsx]:
```suggestion
                <span className="font-bold">{`You searched for "${search}"`}</span>
```

**Review by agustinmrgo — APPROVED** [review]

File: src/components/MobileFilters/MobileFilters.tsx (hunk 1 above)
Comment (DJA89) [reply src/components/MobileFilters/MobileFilters.tsx:10]: It doesn't like it when I don't add them, I imagined that a fragment was needed when there's only text as text alone is not correct JSX syntax I don't think

image

File: src/app/(authenticated)/products/DesktopProducts.tsx (hunk 2 above)
Comment (DJA89) [reply src/app/(authenticated)/products/DesktopProducts.tsx]: Oops, that weird thing was probably added by the linter when I committed, hadn't caught it thanks. I added the empty space inside the span, but not the whole phrase as I only want the search term to be bold.

**Pull request merged**